ddl/column.go

```diff
@@ -518,15 +554,35 @@ func onDropColumn(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		// write only -> delete only
 		job.SchemaState = model.StateDeleteOnly
 		colInfo.State = model.StateDeleteOnly
+		if len(idxInfos) > 0 {
+			setIndicesState(tblInfo, idxInfos, model.StateDeleteOnly)
+		}
 		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != colInfo.State)
 	case model.StateDeleteOnly:
 		// delete only -> reorganization
 		job.SchemaState = model.StateDeleteReorganization
 		colInfo.State = model.StateDeleteReorganization
+		if len(idxInfos) > 0 {
+			setIndicesState(tblInfo, idxInfos, model.StateDeleteReorganization)
+		}
 		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != colInfo.State)
 	case model.StateDeleteReorganization:
 		// reorganization -> absent
 		// All reorganization jobs are done, drop this column.
+		indexIDs := make([]int64, 0, len(idxInfos))
+		if len(idxInfos) > 0 {
+			newIndices := make([]*model.IndexInfo, 0, len(tblInfo.Indices))
+			for _, idx := range tblInfo.Indices {
+				if !indexInfoContains(idx.ID, idxInfos) {
+					newIndices = append(newIndices, idx)
+				}
+			}
+			tblInfo.Indices = newIndices
+			for _, indexInfo := range idxInfos {
+				indexIDs = append(indexIDs, indexInfo.ID)
+			}
+		}
```

Comment:
Could we extract line573-584 into a function？It seems that both `onDropColumn` and `onDropColumns` use it.